pkg/buildclient/messages.go

```diff
@@ -229,8 +230,22 @@ func (m *WebsocketMessages) Recv() (<-chan *Message, func()) {
 }
 
 func (m *WebsocketMessages) Send(msg *Message) error {
-	logrus.Tracef("Send build message %s", msg)
+	logrus.Tracef("Send build message %s", redact(msg))
 	m.lock.Lock()
 	defer m.lock.Unlock()
 	return m.conn.WriteJSON(msg)
 }
+
+// redact removes sensitive information from a Message.
+// Use this before logging a message.
+func redact(msg *Message) *Message {
+	redacted := z.Dereference(msg)
```

Comment:
This is really a gratuitous use of z.Deference, don't use this in place of a `if msg != nil`.  Code tricks obscure the intention and decrease readibility. z.Dereference is mostly acceptable when dealing it native types (*int, *string, *bool) because they are very unnatural to interact with, but a struct pointer shouldn't fall into that bucket.

Reply:
That's reasonable, I'll amend this to

```go
var redacted Message
if msg != nil {
    redacted = *msg
}
```

and update the godocs for `z.Dereference` to prevent future misuse

Reply:
Done.

https://github.com/acorn-io/z/pull/6